tests/mr/host/mr_ref_tests.cpp

```diff
@@ -0,0 +1,262 @@
+
+  std::deque<allocation> allocations;
+
+  constexpr std::size_t num_allocations{100};
+  for (std::size_t i = 0; i < num_allocations; ++i) {
+    std::size_t allocation_size = size_distribution(generator);
+    EXPECT_NO_THROW(allocations.emplace_back(this->ref.allocate(allocation_size), allocation_size));
+    auto new_allocation = allocations.back();
+    EXPECT_NE(nullptr, new_allocation.ptr);
+    EXPECT_TRUE(is_aligned(new_allocation.ptr));
+
+    bool const free_front{free_distribution(generator) == free_distribution.max()};
+
+    if (free_front) {
+      auto front = allocations.front();
+      EXPECT_NO_THROW(this->ref.deallocate(front.ptr, front.size));
+      allocations.pop_front();
+    }
+  }
+  // free any remaining allocations
+  for (auto alloc : allocations) {
+    EXPECT_NO_THROW(this->ref.deallocate(alloc.ptr, alloc.size));
+    allocations.pop_front();
+  }
+}
+
+static constexpr std::size_t MinTestedAlignment{16};
+static constexpr std::size_t MaxTestedAlignment{4096};
+static constexpr std::size_t TestedAlignmentMultiplier{2};
+static constexpr std::size_t NUM_TRIALS{100};
+
+TYPED_TEST(MRRefTest, AlignmentTest)
+{
+  std::default_random_engine generator(0);
+  constexpr std::size_t MAX_ALLOCATION_SIZE{10 * size_mb};
+  std::uniform_int_distribution<std::size_t> size_distribution(1, MAX_ALLOCATION_SIZE);
+
+  for (std::size_t num_trials = 0; num_trials < NUM_TRIALS; ++num_trials) {
+    for (std::size_t alignment = MinTestedAlignment; alignment <= MaxTestedAlignment;
+         alignment *= TestedAlignmentMultiplier) {
+      auto allocation_size = size_distribution(generator);
+      void* ptr{nullptr};
+      EXPECT_NO_THROW(ptr = this->ref.allocate(allocation_size, alignment));
+      EXPECT_TRUE(is_aligned(ptr, alignment));
+      EXPECT_NO_THROW(this->ref.deallocate(ptr, allocation_size, alignment));
+    }
+  }
+}
+
+TYPED_TEST(MRRefTest, UnsupportedAlignmentTest)
+{
+  std::default_random_engine generator(0);
+  constexpr std::size_t MAX_ALLOCATION_SIZE{10 * size_mb};
+  std::uniform_int_distribution<std::size_t> size_distribution(1, MAX_ALLOCATION_SIZE);
+
+  for (std::size_t num_trials = 0; num_trials < NUM_TRIALS; ++num_trials) {
+    for (std::size_t alignment = MinTestedAlignment; alignment <= MaxTestedAlignment;
+         alignment *= TestedAlignmentMultiplier) {
+      auto allocation_size = size_distribution(generator);
+      void* ptr{nullptr};
+      // An unsupported alignment (like an odd number) should result in an
+      // alignment of `alignof(std::max_align_t)`
+      auto const bad_alignment = alignment + 1;
+      EXPECT_NO_THROW(ptr = this->ref.allocate(allocation_size, bad_alignment));
+      EXPECT_TRUE(is_aligned(ptr, alignof(std::max_align_t)));
+      EXPECT_NO_THROW(this->ref.deallocate(ptr, allocation_size, bad_alignment));
+    }
+  }
+}
+
+TEST(PinnedResource, isPinned)
+{
+  rmm::mr::pinned_memory_resource mr;
+  cuda::mr::resource_ref<cuda::mr::host_accessible> ref{mr};
```

Comment:
suggestion: It seems like we're missing a test that verifies `pinned_memory_resource` also satisfies `device_accessible`.

Reply:
There is a static assert right after the pinned_memory_resource definition